File: C# part 2/Arrays/04.MaximalSequence/MaximalSequence.cs
/*
 Problem 4. Maximal sequence

    Write a program that finds the maximal sequence of equal elements in an array.

 */
namespace _04.MaximalSequence
{
    using System;

    class MaximalSequence
    {
        static void Main()
        {
            Console.Write("Enter array length: ");
            int length = int.Parse(Console.ReadLine());

            int[] sequence = new int[length];
            Console.WriteLine("Enter integer values");
            for (int i = 0; i < sequence.Length; i++)
            {
                Console.Write("Array[{0}] = ", i);
                string input = Console.ReadLine();
                int number;
                bool isNumber = int.TryParse(input, out number);
                if (isNumber)
                {
                    sequence[i] = number;
                }
                else
                {
                    Console.WriteLine("Input string was not in a correct format");
                }
            }

            int startElement = 0;
            int sequenceCount = 1;
            int bestStart = 0;
            int bestSequence = 0;

            for (int i = 0; i < sequence.Length - 1; i++)
            {
                if (sequence[i] != sequence[i + 1])
                {
                    startElement = sequence[i + 1];
                    sequenceCount = 1;
                }
                if (sequence[i] == sequence[i + 1])
                {
                    sequenceCount++;
                    if (bestSequence < sequenceCount)
                    {
                        bestSequence = sequenceCount;
                        bestStart = sequence[i + 1];
                    }
                }
            }

            int[] newSequence = new int[bestSequence];
            for (int i = 0; i < newSequence.Length; i++)
            {
                newSequence[i] = bestStart;
            }
            Console.WriteLine("Maximal sequence");
            for (int i = 0; i < newSequence.Length; i++)
            {
                Console.Write("{0} ", newSequence[i]);
            }
            Console.WriteLine();
        }
    }
}

File: JavaScript Fundamentals/Javascript Arrays/scripts/maximalSequence.js
jsConsole.writeLine('--------------------------------------------------------------------');
jsConsole.writeLine('Problem 3: Write a script that finds the maximal sequence of equal elements in an array.')
jsConsole.writeLine('--------------------------------------------------------------------');
console.log('--------------------------------------------------------------------');
console.log('Problem 3: Write a script that finds the maximal sequence of equal elements in an array.')
console.log('--------------------------------------------------------------------');

jsConsole.writeLine('Example: [2, 1, 1, 2, 3, 3, 2, 2, 2, 1]');
console.log('Example: [2, 1, 1, 2, 3, 3, 2, 2, 2, 1]');

var numbers = [2, 1, 1, 2, 3, 3, 2, 2, 2, 1];
var currentCounter = 1;
var number = numbers[0];
var currentNumber = 0;
var maxSequence = 1;

for (var i = 0; i < numbers.length - 1; i++) {
    if (numbers[i] === numbers[i + 1]) {
        currentNumber = numbers[i];
        currentCounter++;
    }
    if (numbers[i] !== numbers[i + 1]) {
        currentCounter = 1;
    }
    if (maxSequence < currentCounter) {
        maxSequence = currentCounter;
        number = currentNumber;
    }
}

for (var i = 0; i < maxSequence; i++) {
    jsConsole.write(number + ' ');
    console.log(number + ' ');
}
jsConsole.writeLine();